fv3core/utils/grid.py

```diff
@@ -419,6 +420,7 @@ def horizontal_starts_from_shape(self, shape):
             return 0, 0
 
 
+@functools.lru_cache
```

Comment:
sweet

Reply:
Doesn't this assume that the object is immutable? Should we be following https://stackoverflow.com/a/4828108 to ensure this?

Reply:
no, this just requires hash-able arguments